Binary_search/search_in_rotated_array.cpp

```diff
@@ -0,0 +1,46 @@
+#include <bits/stdc++.h>
+
+#define lli long long int
+
+using namespace std;
+
+lli findRotEle(lli arr[], lli l, lli r, lli key)
+{
+    if(r >= l)
+    {
+        lli mid = (l + r)/2;
+        if(arr[mid] == key)
+            return mid;
+        if(arr[mid] <= arr[r])
+        {
+            if(arr[mid] < key and arr[r] >= key)
+                return findRotEle(arr, mid+1, r, key);
+            else
+                return findRotEle(arr, l, mid-1, key);
+        }
+        else
+        {
+            if(key >= arr[l] and arr[mid] > key)
+                return findRotEle(arr, l, mid-1, key);
+            else
+                return findRotEle(arr, mid+1, r, key);
+        }
+    }
+    return -1;
+}
+
+int main()
+{
+    lli test;
+    cin >> test;
```

Comment:
Hi, @hinduBale thanks for making changes. Can you add a comment in this file also which inputs are being taken, or print it on the console?